gapic/src/main/com/google/gapid/views/TextureList.java

```diff
@@ -0,0 +1,551 @@
+      @Override
+      public boolean select(Viewer viewer, Object parentElement, Object element) {
+        return !((Data)element).deleted;
+      }
+    };
+
+    textureTable.getTable().addListener(SWT.Selection, e -> updateSelection());
+    showDeleted.addListener(SWT.Selection, e -> {
+      if (showDeleted.getSelection()) {
+        textureTable.removeFilter(filterDeleted);
+      } else {
+        textureTable.addFilter(filterDeleted);
+      }
+    });
+  }
+
+  private static void initTextureSelector(TableViewer viewer, ImageProvider imageProvider) {
+    viewer.setContentProvider(ArrayContentProvider.getInstance());
+    viewer.setLabelProvider(new ViewLabelProvider(imageProvider));
+
+    sorting(viewer,
+        createTableColumn(viewer, "Type", Data::getType,
+            Comparator.comparing(Data::getType)),
+        createTableColumn(viewer, "ID", Data::getId, d -> imageProvider.getImage(d),
+            (d1, d2) -> UnsignedLongs.compare(d1.getSortId(), d2.getSortId())),
+        createTableColumn(viewer, "Name", Data::getLabel,
+            Comparator.comparing(Data::getLabel)),
+        createTableColumn(viewer, "Width", Data::getWidth,
+            Comparator.comparingInt(Data::getSortWidth)),
+        createTableColumn(viewer, "Height", Data::getHeight,
+            Comparator.comparingInt(Data::getSortHeight)),
+        createTableColumn(viewer, "Depth", Data::getDepth,
+            Comparator.comparingInt(Data::getSortDepth)),
+        createTableColumn(viewer, "Layers", Data::getLayers,
+            Comparator.comparingInt(Data::getSortLayers)),
+        createTableColumn(viewer, "Levels", Data::getLevels,
+            Comparator.comparingInt(Data::getSortLevels)),
+        createTableColumn(viewer, "Format", Data::getFormat,
+            Comparator.comparing(Data::getFormat)));
+  }
+
+  @Override
+  public Control getControl() {
+    return this;
+  }
+
+  @Override
+  public void reinitialize() {
+    onCaptureLoadingStart(false);
+    updateTextures(true);
+  }
+
+  @Override
+  public void onCaptureLoadingStart(boolean maintainState) {
+    clear();
+  }
+
+  @Override
+  public void onCaptureLoaded(Loadable.Message error) {
+    if (error != null) {
+      clear();
+    }
+  }
+
+  @Override
+  public void onResourcesLoaded() {
+    updateTextures(true);
+  }
+
+  @Override
+  public void onCommandsSelected(CommandIndex path) {
+    updateTextures(false);
+  }
+
+  @Override
+  public void onTextureFollowed(Service.Resource resource) {
+    TableItem[] items = textureTable.getTable().getItems();
+    for (int i = 0; i < items.length; i++) {
+      Data d = (Data)(items[i].getData());
+      if (d.info.getID().equals(resource.getID())) {
+        textureTable.getTable().setSelection(items[i]);
+        break;
+      }
+    }
+  }
+
+  private void updateTextures(boolean resourcesChanged) {
+    if (!models.resources.isLoaded()) {
+      // TODO: show capture loading message
```

Comment:
This needs to be done before it's submitted.

Reply:
Done